core/src/main/scala/dagr/core/execsystem/TaskManager.scala

```diff
@@ -70,7 +71,8 @@ case class SystemResources(cores: Cores, systemMemory: Memory, jvmMemory: Memory
 object TaskManagerDefaults extends LazyLogging {
   def defaultTaskManagerResources: SystemResources = {
     val resources = SystemResources(cores=None, totalMemory=None) // Let the apply method figure it all out
-    logger.info("Defaulting System Resources to " + resources.cores.value + " cores and " + Resource.parseBytesToSize(resources.systemMemory.value) + " memory")
+    val cores     = TaskManager.roundCoresForLogging(resources.cores)
```

Comment:
why not just `f"${resources.cores}%.2f"`  here and everywhere?

Reply:
Could do! If I used `printf`-style formatting then you would **always** have 2 decimals, even if you didn't need them. For example, with `printf`-style formatting, if you had 4 cores exactly, you would log that as `4.00` whereas with the current implementation, it would be simply `4`.

Reply:
Up to you, would you rather see trailing zeros?

Reply:
Actually, I realized I'm using Doubles which print with at least one trailing zero. So using string formatting like you've suggested would most likely be the cleanest approach all around.

Reply:
I didn’t think of that, so a happy accident all around!

Reply:
I went with printf-style formatting, 2 decimals places.